mlflow/spark.py

```diff
@@ -235,6 +239,9 @@ def log_model(
     try:
         spark_model.save(posixpath.join(model_dir, _SPARK_MODEL_PATH_SUB))
     except Py4JError:
+        if databricks_utils.is_mlflowdbfs_available():
```

Comment:
We also need to check that the run's artifact root URI is of the form `dbfs:/databricks/mlflow-tracking`, which we can accomplish by fetching the artifact repo associated with the run and verifying that it's an instance of `DatabricksArtifactRepository`.

If we don't perform this check, model logging will fail in MLR / DBR for experiments with custom artifact root locations, such as `s3://...`, `dbfs:/my/custom/root/locaton`, etc.